eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentEvents.scala

```diff
@@ -251,11 +251,13 @@ object PaymentFailure {
 
 }
 
-case class PathFindingExperimentMetrics(amount: MilliSatoshi,
+case class PathFindingExperimentMetrics(paymentHash: ByteVector32,
+                                        amount: MilliSatoshi,
                                         fees: MilliSatoshi,
                                         status: String,
                                         duration: FiniteDuration,
                                         timestamp: TimestampMilli,
                                         isMultiPart: Boolean,
                                         experimentName: String,
-                                        recipientNodeId: PublicKey)
+                                        recipientNodeId: PublicKey,
+                                        routingHints: Set[PublicKey])
```

Comment:
It's not obvious at all what `routingHints` means here. Looking at impl I think we put the first node of every available route hint. 

But nothing prevents us to not use hints at all (destination could be a public node) or to only use an arbitrary tail of a provided route hint right?

Why don't we just log the full routing hint as a json? It would make everything simpler and you could decide at query time what you want to do.

Reply:
We've previously discussed this here for context: https://github.com/ACINQ/eclair/pull/2242#discussion_r858490598
I also think that it makes sense to log all routing hints (this way we have all potential information, even if some turns out to be irrelevant).

Reply:
> We've previously discussed this here for context: https://github.com/ACINQ/eclair/pull/2242#discussion_r858490598

I somehow missed it.

> I'm assuming that this field will contain 0 or 1 node id

@thomash-acinq This assumption might hold true now but maybe not in the future, so better make it at query time I think.

Here is what a query would look like:
```sql
SELECT jsonb_array_elements(hints) #> '{0,nodeId}'
"02413957815d05abb7fc6d885622d5cdc5b7714db1478cb05813a8474179b83c5c"
"03933884aaf1d6b108397e5efe5c86bcf2d8ca8d2f700eda99db9214fc2712b134"
```

Assuming that the data is:
```sql
WITH data AS (SELECT '[
    [
      {
        "nodeId": "02413957815d05abb7fc6d885622d5cdc5b7714db1478cb05813a8474179b83c5c",
        "shortChannelId": "1975837x88x0",
        "feeBase": 1000,
        "feeProportionalMillionths": 100,
        "cltvExpiryDelta": 144
      },
      {
        "nodeId": "0225ff2ae6a3d9722b625072503c2f64f6eddb78d739379d2ee55a16b3b0ed0a17",
        "shortChannelId": "123454x42x2",
        "feeBase": 1000,
        "feeProportionalMillionths": 100,
        "cltvExpiryDelta": 144
      }
    ],
    [
      {
        "nodeId": "03933884aaf1d6b108397e5efe5c86bcf2d8ca8d2f700eda99db9214fc2712b134",
        "shortChannelId": "1976152x25x0",
        "feeBase": 1000,
        "feeProportionalMillionths": 100,
        "cltvExpiryDelta": 144
      }
    ]
  ]'::jsonb as hints)
```

Reply:
Done, I now put everything and will let Postgres do its JSON magic